### PR TITLE
feat(std/node): Port most of node errors

### PR DESCRIPTION
**Pull request by Soremwar**
Based on #7569

### CODE REVIEW

File: std/node/_stream/async_iterator.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+// Add Duplex streams
+type IterableStreams = Stream | Readable;
+
+type IterableItem = Buffer | string | Uint8Array | undefined;
+type ReadableIteratorResult = IteratorResult<IterableItem>;
+
+function initIteratorSymbols(
+  o: ReadableStreamAsyncIterator,
+  symbols: symbol[],
+) {
```
Comment (bartlomieju) [inline std/node/_stream/async_iterator.ts:24]: @Soremwar should this code be included in this PR?

Comment (Soremwar) [reply std/node/_stream/async_iterator.ts:24]: Nothing inside _stream will affect be included in this PR. It will be rebased once #7569 is merged

**Comment by bartlomieju** [comment]
@Soremwar please rebase

**Comment by Soremwar** [comment]
@bartlomieju Done

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**